chia/wallet/did_wallet/did_wallet_puzzles.py

```diff
@@ -21,49 +21,85 @@
 
 def create_innerpuz(
     p2_puzzle: Program,
-    identities: List[bytes],
+    recovery_list: List[bytes],
     num_of_backup_ids_needed: uint64,
-    singleton_id: bytes32,
+    launcher_id: bytes32,
     metadata: Program = Program.to([]),
 ) -> Program:
-    backup_ids_hash = Program(Program.to(identities)).get_tree_hash()
-    singleton_struct = Program.to((SINGLETON_MOD_HASH, (singleton_id, LAUNCHER_PUZZLE_HASH)))
+    """
+    Create DID inner puzzle
+    :param p2_puzzle: Standard P2 puzzle
+    :param recovery_list: A list of DIDs used for the recovery
+    :param num_of_backup_ids_needed: Need how many DIDs for the recovery
+    :param launcher_id: ID of the launch coin
+    :param metadata: DID customized metadata
+    :return: DID inner puzzle
+    """
+    backup_ids_hash = Program(Program.to(recovery_list)).get_tree_hash()
+    singleton_struct = Program.to((SINGLETON_MOD_HASH, (launcher_id, LAUNCHER_PUZZLE_HASH)))
     return DID_INNERPUZ_MOD.curry(p2_puzzle, backup_ids_hash, num_of_backup_ids_needed, singleton_struct, metadata)
 
 
 def get_inner_puzhash_by_p2(
     p2_puzhash: bytes32,
-    identities: List[bytes],
+    recovery_list: List[bytes],
```

Comment:
is the `recovery_list` a list of `bytes32`?

Reply:
I think it is, currently. Matt defined this as bytes, not sure if it can be something else in the future.